src/hugr/hugrmut.rs

```diff
@@ -221,38 +116,154 @@ impl HugrMut {
     ///
     ///  - If the sibling node does not have a parent.
     ///  - If the attachment would introduce a cycle.
-    pub fn add_op_after(
+    fn add_op_after(&mut self, sibling: Node, op: impl Into<OpType>) -> Result<Node, HugrError>;
+
+    /// Build the HUGR, returning an error if the graph is not valid.
+    fn finish(self) -> Result<Hugr, ValidationError>;
```

Comment:
This shouldn't be in the trait.
We can already dereference; finishing the hugr should be done by whatever builder struct you are using (or if you already have a `Hugr` just call `validate`).

Reply:
Yes, I meant to call this out, sorry - it looked a bit dubious. Removing it entirely has meant that many of the `impl<T: HugrMut> Container for SomeBuilder<T>` have become `impl Container for SomeBuilder<Hugr>` which is to say, you can only call `.finish_hugr` when building a whole Hugr. Given that we are not doing mutable subregions yet, this is good :)